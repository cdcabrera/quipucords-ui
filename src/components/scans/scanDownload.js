import React from 'react';
import PropTypes from 'prop-types';
import { Dropdown, Icon, MenuItem } from 'patternfly-react';
import { connect, reduxActions, reduxTypes, store } from '../../redux';
import { helpers } from '../../common/helpers';

class ScanDownload extends React.Component {
  static notifyDownloadStatus(error, results) {
    if (error) {
      store.dispatch({
        type: reduxTypes.toastNotifications.TOAST_ADD,
        alertType: 'error',
        header: 'Error',
        message: helpers.getMessageFromResults(results).message
      });
    } else {
      store.dispatch({
        type: reduxTypes.toastNotifications.TOAST_ADD,
        alertType: 'success',
        message: <span>Report downloaded.</span>
      });
    }
  }

  onDetailedDownload = () => {
    const { getReportDetailsCsv, downloadId } = this.props;

    getReportDetailsCsv(downloadId).then(
      () => ScanDownload.notifyDownloadStatus(false),
      error => ScanDownload.notifyDownloadStatus(true, error.message)
    );
  };

  onSummaryDownload = () => {
    const { getReportSummaryCsv, downloadId } = this.props;

    getReportSummaryCsv(downloadId).then(
      () => ScanDownload.notifyDownloadStatus(false),
      error => ScanDownload.notifyDownloadStatus(true, error.message)
    );
  };

  render() {
    const { downloadId, getReportDetailsCsv, getReportSummaryCsv, ...props } = this.props;

    return (
      <React.Fragment>
        <Dropdown id={helpers.generateId()} {...props}>
          <Dropdown.Toggle useAnchor className="btn-link">
            <Icon type="fa" name="download" />
          </Dropdown.Toggle>
          <Dropdown.Menu>
            <MenuItem eventKey="1" onClick={() => this.onSummaryDownload()}>
              Summary Report
            </MenuItem>
            <MenuItem eventKey="2" onClick={() => this.onDetailedDownload()}>
              Detailed Report
            </MenuItem>
          </Dropdown.Menu>
        </Dropdown>
      </React.Fragment>
    );
  }
}

ScanDownload.propTypes = {
  downloadId: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  getReportDetailsCsv: PropTypes.func,
  getReportSummaryCsv: PropTypes.func
};

ScanDownload.defaultProps = {
  getReportDetailsCsv: helpers.noop,
  getReportSummaryCsv: helpers.noop
};

const mapDispatchToProps = dispatch => ({
  getReportDetailsCsv: id => dispatch(reduxActions.reports.getReportDetailsCsv(id)),
  getReportSummaryCsv: (id, query) => dispatch(reduxActions.reports.getReportSummaryCsv(id, query))
});

const mapStateToProps = () => ({});

const ConnectedScanDownload = connect(
  mapStateToProps,
  mapDispatchToProps
)(ScanDownload);

export { ConnectedScanDownload as default, ConnectedScanDownload, ScanDownload };
