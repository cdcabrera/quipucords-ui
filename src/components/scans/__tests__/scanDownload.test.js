import React from 'react';
import configureMockStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import { ConnectedScanDownload, ScanDownload } from '../scanDownload';

describe('ScanDownload Component', () => {
  const generateEmptyStore = (obj = {}) => configureMockStore()(obj);

  it('should render a connected component', () => {
    const store = generateEmptyStore({});

    const props = {
      downloadId: 1
    };

    const component = shallow(<ConnectedScanDownload {...props} />, {
      context: { store }
    });

    expect(component).toMatchSnapshot('connected');
  });

  it('should render a non-connected component', () => {
    const props = {
      downloadId: 1
    };

    const component = mount(<ScanDownload {...props} />);

    expect(component.render()).toMatchSnapshot('non-connected');
  });
});
