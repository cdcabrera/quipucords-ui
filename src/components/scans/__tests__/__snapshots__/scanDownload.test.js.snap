// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScanDownload Component should handle custom children: custom 1`] = `
<button
  class="btn btn-default"
  id="generatedid-"
  title="Download"
  type="button"
>
  lorem ipsum
</button>
`;

exports[`ScanDownload Component should have an optional tooltip: tooltip 1`] = `
<ScanDownload
  downloadId={1}
  downloadName={null}
  getReportsDownload={[Function]}
  tooltip="Lorem ipsum dolor sit"
>
  <Tooltip
    delayShow={500}
    id={null}
    placement="top"
    popover={null}
    rootClose={true}
    tooltip="Lorem ipsum dolor sit"
    trigger={
      Array [
        "hover",
      ]
    }
  >
    <OverlayTrigger
      defaultOverlayShown={false}
      delayShow={500}
      overlay={
        <Tooltip
          bsClass="tooltip"
          id="generatedid-"
          placement="right"
        >
          Lorem ipsum dolor sit
        </Tooltip>
      }
      placement="top"
      rootClose={true}
      trigger={
        Array [
          "hover",
        ]
      }
    >
      <span
        onClick={null}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          id="generatedid-"
          onClick={[Function]}
          title="Download"
        >
          <button
            className="btn btn-default"
            disabled={false}
            id="generatedid-"
            onClick={[Function]}
            title="Download"
            type="button"
          >
            Download
          </button>
        </Button>
      </span>
    </OverlayTrigger>
  </Tooltip>
</ScanDownload>
`;

exports[`ScanDownload Component should render a connected component: connected 1`] = `
<button
  class="btn btn-default"
  id="generatedid-"
  title="Download"
  type="button"
>
  Download
</button>
`;

exports[`ScanDownload Component should render a non-connected component: non-connected 1`] = `
<button
  class="btn btn-default"
  id="generatedid-"
  title="Download"
  type="button"
>
  Download
</button>
`;
